### PR TITLE
S3 upload

### PR DESCRIPTION
**Pull request by Sofistication**
`uploads#create` now recieves a file from the client and uploads it to s3.
Mongo documents now actually save the url of the uploaded file.

Adds `multer` as a dependency

solves #3

### CODE REVIEW

File: app/controllers/uploads.js [hunk 1]
```diff
@@ -3,11 +3,15 @@
...
 const models = require('app/models');
 const Upload = models.upload;
+const s3Upload = require('lib/s3-upload');
 
 const authenticate = require('./concerns/authenticate');
 const setUser = require('./concerns/set-current-user');
 const setModel = require('./concerns/set-mongoose-model');
 
+const multer = require('multer');
+const multerUploader = multer({ dest: '/tmp/' });
```
Comment (ddobson) [inline app/controllers/uploads.js]: Lookup Multer limits for file size. https://www.npmjs.com/package/multer

File: app/controllers/uploads.js [hunk 2]
```diff
@@ -24,16 +28,31 @@ const show = (req, res) => {
...
-    _owner: req.user._id,
-  });
-  Upload.create(upload)
-    .then(upload =>
-      res.status(201)
-        .json({
-          upload: upload.toJSON({ virtuals: true, user: req.user }),
-        }))
-    .catch(next);
+  console.log('req is ', req);
```
Comment (ddobson) [inline app/controllers/uploads.js]: Remove log and comments, plzzzz.

File: app/controllers/uploads.js [hunk 3]
```diff
@@ -24,16 +28,31 @@ const show = (req, res) => {
...
+  // });
+  // Upload.create(upload)
+  //   .then(upload =>
+  //     res.status(201)
+  //       .json({
+  //         upload: upload.toJSON({ virtuals: true, user: req.user }),
+  //       }))
+  //   .catch(next);
+
+  s3Upload(req.file)
```
Comment (ddobson) [inline app/controllers/uploads.js]: You can't just pass a file here. You need to send options or you need to change the arguments sent to the lib function.

File: app/controllers/uploads.js [hunk 4]
```diff
@@ -24,16 +28,31 @@ const show = (req, res) => {
...
+    .then((s3Response) => {
+      return Upload.create({
+        _owner: req.user._id,
+        fileName: req.file.originalName,
+        url: s3Response.location,
+        tag: req.body.tag,
```
Comment (ddobson) [inline app/controllers/uploads.js]: Please change `tag` to `tags` here and in the model.

File: app/controllers/uploads.js (hunk 1 above)
Comment (Sofistication) [inline app/controllers/uploads.js]: 7a1bfce

File: app/controllers/uploads.js (hunk 3 above)
Comment (Sofistication) [inline app/controllers/uploads.js]: 81ce5e0

File: app/controllers/uploads.js (hunk 4 above)
Comment (Sofistication) [inline app/controllers/uploads.js]: 36bace8

**Pull request merged**